### PR TITLE
[configcat] Update to version 4.0.3

### PR DESCRIPTION
**Pull request by kp-cat**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/configcat/vcpkg.json [hunk 1]
```diff
@@ -36,6 +36,12 @@
           "platform": "linux"
         }
       ]
+    },
+    "sha": {
+      "description": "Use built-in hash-library",
```
Comment (dg0yt) [inline ports/configcat/vcpkg.json:41]: IIUC it is
```suggestion
      "description": "Use external hash-library",
```

**Comment by Cheney-W** [comment]
@kp-cat Could you please apply dg0yt's suggestion?

File: ports/configcat/vcpkg.json (hunk 1 above)
Comment (Cheney-W) [reply ports/configcat/vcpkg.json:41]: @kp-cat 
Thanks for your reply! Please click the `Ready for review` button if you think this pr is complete.

Comment (dg0yt) [reply ports/configcat/vcpkg.json:41]: From my what I understand now, all cases are "external" (to configcat), not "builtin".

Comment (kp-cat) [reply ports/configcat/vcpkg.json]: Hey @dg0yt,
I changed the descriptions (removed external/builtin and use just the library name).

**Review by Cheney-W — APPROVED** [review]

**Pull request merged**